### PR TITLE
feat: Enhance renovate configuration

### LINKED ISSUE #1: Improve renovate configuration
The current renovate config is very basic, let's enrich this based on lessons learned from other sample applications.

### PR DESCRIPTION
**Pull request by grayside**
🛠️ Fixes #1 

Let's modify the defaults as follows:
* Switch to json5 to support code comments
* Emphasize presets where possible to minimize amount of configuration
* Group by language to enable single-SME reviews
* Include lessons learned about tightly-knit ecosystems such as Playwright and OTel
* Schedule on a 2 week sprint basis.
* Add labels as suitable, and reviewer guidance to PR descriptions
* Move to `.github/`

### CODE REVIEW

File: .github/renovate.json5 [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+//
+// See https://docs.renovatebot.com/configuration-options/
+// See https://json5.org/ for JSON5 syntax
+{
+  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
+  "extends": [
+    "config:base", // https://docs.renovatebot.com/presets-config/#configbase
+    ":semanticCommits", // https://docs.renovatebot.com/presets-default/#semanticcommits
+    ":ignoreUnstable", // https://docs.renovatebot.com/presets-default/#ignoreunstable
+    "group:allNonMajor", // https://docs.renovatebot.com/presets-group/#groupallnonmajor
```
Comment (averikitsch) [inline .github/renovate.json5:11]: This is probably good for one off sample repos or JSS repos but not LANG-docs-samples repos

File: .github/renovate.json5 [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+      // Increases build repeatability, image cache use, and supply chain security.
+      // "pinDigests": true,
+    },
+    {
+       "groupName": "GitHub Actions",
+       "matchManagers": ["github-actions"],
+       "pinDigests": true,
+    },
+
+    // Language groupings.
```
Comment (averikitsch) [inline .github/renovate.json5:80]: Does this section of matching do the same things as `group:allNonMajor`?

File: .github/renovate.json5 [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+      "matchDepTypes": ["devDependencies"],
+      "separateMajorMinor": false,
+    },
+    {
+      "groupName": "Terraform",
+      "matchManagers": ["terraform"],
+      "separateMajorMinor": false,
+    },
+
+    // Updating coupled ecosystems.
```
Comment (averikitsch) [inline .github/renovate.json5:119]: Should we have a per language renovate or do we want to start collecting more packages here?

File: .github/renovate.json5 (hunk 1 above)
Comment (grayside) [reply .github/renovate.json5:11]: golang-samples is even more aggressive, grouping everything except majors: https://github.com/GoogleCloudPlatform/golang-samples/blob/main/.github/renovate.json

I didn't realize this repo was potentially to springboard new mono-repos. Do you think we should leave this in but add a comment guiding when to remove?

File: .github/renovate.json5 (hunk 2 above)
Comment (grayside) [reply .github/renovate.json5:80]: group:allNonMajor is a fallback for unspecific languages and things like lockfile updates (the latter which renovate recommends to handle separately from dependency updates).

This section groups nonmajor per language, so for example, all Java minor & patch updates should be grouped, but C++ would fall back into a general catch all. This way we default to a fairly aggressive grouping approach, but try to line up reviews with one language at a time to expedite the process.

File: .github/renovate.json5 (hunk 3 above)
Comment (grayside) [reply .github/renovate.json5:119]: Could you expand on the problem? Per language runs counter to the goal I had in supporting multi-lang apps so hoping to understand more and see what balance to change.

Something else we could do is find a way to host some presets, and then the renovate config in the template could become smaller, and more comment heavy around tailoring for languages in use.

File: .github/renovate.json5 (hunk 1 above)
Comment (averikitsch) [reply .github/renovate.json5:11]: We can leave it in. Mono repo support is a secondary goal

File: .github/renovate.json5 (hunk 3 above)
Comment (averikitsch) [reply .github/renovate.json5:119]: My original review had the idea we would port some learns from here to the mono repos. Now I am declaring that a secondary goal.

**Review by averikitsch — APPROVED** [review]

**Pull request merged**